docs/conf.py

```diff
@@ -144,6 +144,7 @@
         "custom-template",
     ],  # This ensures we test for custom sidebars
     "demo/no-sidebar": [],  # Test what page looks like with no sidebar items
+    "index": [],
```

Comment:
even after this change, `nox -s test` is failing for me locally:

```
assert not index_html.select(".bd-sidebar-primary")
```

@choldgraf any idea why?

Reply:
Hmmm, could it be because of one of the test sites?

https://github.com/pydata/pydata-sphinx-theme/blob/94a9e8c7f1bf843172622bc38928658d3537347a/tests/sites/base/conf.py#L23

Reply:
I don't think that explains it... the failing test line is targeting `index_html` not `subpage_html` (which is what `section1/index` is called in the test)

https://github.com/pydata/pydata-sphinx-theme/blob/c127146a5afcc94a9522c685a70a49a6d192d10c/tests/test_build.py#L63-L64

Reply:
I can make the test pass by adding `"index": []` to the html_sidebars dict in `tests/sites/base/conf.py`.  But this reflects what I think is undesirable behavior (which I also see with `nox -s docs-live`): namely, the (empty) left sidebar is still there unless I explicitly add `index=[]` to `html_sidebars` in `conf.py`.  I think the behavior you want (and the behavior on current main) is that it should automatically go away when it's empty, right?  So I think we need to put back the removed logic in `sidebar-primary`

Reply:
I think that the suggested logic at the comment below _should_ fix it, because if the sidebars only container the `nav-items` component, and `nav-items` was empty, then it would be removed from `sidebars` and this would then be empty as well.

https://github.com/pydata/pydata-sphinx-theme/pull/777/files#r915581382

Reply:
OK.  I didn't understand that suggestion the first time but now I think I get it.